lib/elixir/lib/module.ex

```diff
@@ -861,6 +861,46 @@ defmodule Module do
     end, tuples)
   end
 
+  @spec make_overridable(module, module) :: :ok
+  def make_overridable(module, behaviour) when is_atom(module) and is_atom(behaviour) do
+    behaviour_definitions = :ets.lookup_element(data_table_for(module), :behaviour, 2)
+
+    error_explanation = cond do
+      !Code.ensure_compiled?(behaviour)                  -> "it was not defined"
+      !function_exported?(behaviour, :behaviour_info, 1) -> "it does not define any callbacks"
+      !:lists.member(behaviour, behaviour_definitions)   -> "its corresponding @behaviour #{inspect(behaviour)} was not specified"
+      true -> nil
+    end
+
+    if error_explanation do
+      raise ArgumentError, "cannot pass module #{inspect(behaviour)} as argument to defoverridable/2 because #{error_explanation}"
+    end
+
+    function_definitions = :ets.match(defs_table_for(module), {{:def, :"$1"}, :"_", :"_", :"_", :"_", :"_"})
+
+    behaviour_callbacks = :lists.map(fn tuple ->
+      normalize_macro_or_function_callback(tuple)
+    end, behaviour.behaviour_info(:callbacks))
+
+    tuples = :lists.filtermap(fn [function_tuple] ->
+      case :lists.member(function_tuple, behaviour_callbacks) do
+        true -> {true, function_tuple}
+        false -> false
+      end
+    end, function_definitions)
```

Comment:
It seems this could be implemented as a comprehension:

    for function_tuple <- definitions_in(module),
        function_tuple in behaviour_callbacks,
        do: function_tuple